src/common/cockpitpipe.c

```diff
@@ -1234,8 +1234,10 @@ spawn_setup (gpointer data)
   prctl (PR_SET_PDEATHSIG, SIGHUP);
 #endif
 
-  if (flags & COCKPIT_PIPE_STDERR_TO_STDOUT)
-    dup2 (1, 2);
+  if (flags & COCKPIT_PIPE_STDERR_TO_STDOUT) {
+    int r = dup2 (1, 2);
+    g_assert (r == 2); /* that should really never fail */
```

Comment:
`g_assert_cmpint()` would be mildly nicer here

Reply:
If I recall correctly, covscan cannot recognize `g_assert_cmpint`, but surely I can change it to that if it is better and then waive it in covscan.

Reply:
Correct, I just waived a false coverity positive yesterday that completely ignored `g_assert_cmpint()`.

Reply:
So what do you think, which would be better? @allisonkarlitskaya @martinpitt

Reply:
I don't really care.  `g_assert_cmpint()` is definitely better, but if we put this here only to make Coverity happy, then perhaps we should actually put something that makes Coverity happy :shrug:

Reply:
Exactly. At the moment I'm not even sure that coverity properly recognizes `g_assert()`. But _cmpint doesn't help here.